Altimeter

DennisGoldfarb opened the pull request:
PR to add the Altimeter model for predicting peptide fragmentation with HCD.

Not be merged yet. The model on zenodo isn't our final version, but it should be enough to test compatibility.

Still to do:
- Finalize model
- Create test output for the final model

DennisGoldfarb commented:
Hey @LLautenbacher this PR is now ready to merge on our end. The core model is in torchscript format because it was faster than ONNX, but our spline evaluation model is ONNX because torchscript doesn't support recursion. Hopefully the tests work on your end. Please let us know if you run into any issues or if we should make changes.

LLautenbacher commented:
Some files are missing. You don't have the option to run the tests locally?

```
I0312 16:09:49.851884 602 pb_stub.cc:323] Failed to initialize Python stub: FileNotFoundError: [Errno 2] No such file or directory: 'Altimeter/Altimeter_2024_filter_isotopes//ion_dictionary.txt'                                                            
                                                                                                                                                                                                                                                              
At:                                                                                                                                                                                                                                                           
  /models/repo/Altimeter_2024_filter_isotopes/1/model.py(175): parseIonDictionary                                                                                                                                                                             
  /models/repo/Altimeter_2024_filter_isotopes/1/model.py(23): initialize                                                                                                                                                                                                                                                                                                                                                                                                                                                    
I0312 16:09:49.875357 602 pb_stub.cc:323] Failed to initialize Python stub: FileNotFoundError: [Errno 2] No such file or directory: 'Altimeter/Altimeter_2024_filter_splines/ion_dictionary.txt'                                                              
                                                                                                                                                                                                                                                              
At:                                                                                                                                                                                                                                                           
  /models/repo/Altimeter_2024_filter_splines/1/model.py(162): parseIonDictionary                                                                                                                                                                              
  /models/repo/Altimeter_2024_filter_splines/1/model.py(24): initialize                                                                                                                                                                                                                                                                                                                                                                                                                                                     
E0312 16:09:50.243824 602 model_lifecycle.cc:626] failed to load 'Altimeter_2024_filter_isotopes' version 1: Internal: FileNotFoundError: [Errno 2] No such file or directory: 'Altimeter/Altimeter_2024_filter_isotopes//ion_dictionary.txt'                 
                                                                                                                                                                                                                                                              
At:                                                                                                                                                                                                                                                           
  /models/repo/Altimeter_2024_filter_isotopes/1/model.py(175): parseIonDictionary                                                                                                                                                                             
  /models/repo/Altimeter_2024_filter_isotopes/1/model.py(23): initialize                                                                                                                                                                                      
                                                                                                                                                                                                                                                              
I0312 16:09:50.243901 602 model_lifecycle.cc:753] failed to load 'Altimeter_2024_filter_isotopes'                                                                                                                                                             
E0312 16:09:50.259230 602 model_lifecycle.cc:626] failed to load 'Altimeter_2024_filter_splines' version 1: Internal: FileNotFoundError: [Errno 2] No such file or directory: 'Altimeter/Altimeter_2024_filter_splines/ion_dictionary.txt'                    
                                                                                                                                                                                                                                                              
At:                                                                                                                                                                                                                                                           
  /models/repo/Altimeter_2024_filter_splines/1/model.py(162): parseIonDictionary                                                                                                                                                                              
  /models/repo/Altimeter_2024_filter_splines/1/model.py(24): initialize      
```

DennisGoldfarb commented:
Hey @LLautenbacher, sorry for taking so long, we had a lot of work to do on Pioneer. I updated the tests and they should work now. They all worked for me locally, albeit I ran the tests on a CPU version of Koina. Hopefully they give the same results on the GPU. I also fleshed out the docs more to follow the format of the other models.

I added a 4th ensemble model called Altimeter_2024_splines_index which is the same as Altimeter_2024_splines except it outputs an annotation index rather than the annotation name as a string. From my testing on a local Triton server, creating the strings took up a lot of time. There's an ion_dictionary.txt file linked in the docs that provides a map of the index to the annotation name. This is the version of the model that Pioneer uses to speed up the library prediction.

Please let me know if you have any issues or requests.

Best,
Dennis

DennisGoldfarb commented:
Yup good call @LLautenbacher. Added them.

DennisGoldfarb commented:
Okay I see there are styling issues, I'll fix those in a bit

DennisGoldfarb commented:
Excellent! Please publish whenever you're ready. Thanks for all your help @LLautenbacher
